packages/signature-controller/src/SignatureController.ts

```diff
@@ -0,0 +1,680 @@
+import EventEmitter from 'events';
+import {
+  MessageManager,
+  MessageParams,
+  MessageParamsMetamask,
+  PersonalMessageManager,
+  PersonalMessageParams,
+  PersonalMessageParamsMetamask,
+  TypedMessageManager,
+  TypedMessageParams,
+  TypedMessageParamsMetamask,
+  AbstractMessageManager,
+  AbstractMessage,
+  MessageManagerState,
+  AbstractMessageParams,
+  AbstractMessageParamsMetamask,
+  OriginalRequest,
+} from '@metamask/message-manager';
+import { ethErrors } from 'eth-rpc-errors';
+import { bufferToHex } from 'ethereumjs-util';
+
+import {
+  BaseControllerV2,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { Patch } from 'immer';
+import {
+  AcceptRequest,
+  AddApprovalRequest,
+  RejectRequest,
+} from '@metamask/approval-controller';
+import { ORIGIN_METAMASK } from '@metamask/controller-utils';
+
+const controllerName = 'SignatureController';
+const methodNameSign = 'eth_sign';
+const methodNamePersonalSign = 'personal_sign';
+const methodNameTypedSign = 'eth_signTypedData';
```

Comment:
These are used as approval types, so they should probably use `ApprovalType` enum.

Reply:
:heavy_plus_sign:

Reply:
Done.